sway-lib-std/src/bytes.sw

```diff
@@ -631,6 +635,15 @@ impl Bytes {
     /// assert(bytes.capacity() == first_length + first_length);
     /// ```
     pub fn append(ref mut self, ref other: self) {
+        if other.len == 0 {
+            return
+        };
+
+        // optimization for when starting with empty bytes and appending to it
+        if self.len == 0 {
+            self.buf.ptr = other.buf.ptr;
+        };
```

Comment:
```suggestion
        if self.len == 0 {
            self.buf.ptr = other.buf.ptr;
            self.buf.cap = other.buf.cap;
            self.len = other.len;
            other.buf.ptr = alloc_bytes(0);
            other.buf.cap = 0;
            other.len = 0;
            return;
        };
```
I was thinking something like this.

By setting the self pointer to the other pointer and then just continuing I think later on you're just copying from the same buffer to itself, byte by byte.  Better to just explicitly move the other buffer to self.

Ideally it'd be less low level than this though.  Like just `self = other; other.clear()` or something.

Reply:
done!